src/TropicalGeometry/curve.jl

```diff
@@ -42,26 +44,487 @@ end
+
+julia> tc = TropicalCurve{min}(IM)
+An abstract tropical curve
+
+julia> coeffs = [0, 1, 1, 1];
+
+julia> dtc = DivisorOnTropicalCurve(tc,coeffs)
+DivisorOnTropicalCurve{min, false}(An abstract tropical curve, [0, 1, 1, 1])
+
+julia> degree(dtc)
+3
+```
+"""
+degree(dtc::DivisorOnTropicalCurve{M, EMB}) where {M, EMB} = sum(coefficients(dtc))
+
+@doc Markdown.doc"""
+    is_effective(dtc::DivisorOnTropicalCurve{M, EMB})              
+
+Check whether a divisor `dtc` on an abstract tropical curve is effective.
+```jldoctest
+julia> IM = IncidenceMatrix([[1,2],[1,3],[1,4],[2,3],[2,4],[3,4]]);
+
+julia> tc = TropicalCurve{min}(IM)
+An abstract tropical curve
+
+julia> coeffs = [0, 1, 1, 1];
+
+julia> dtc = DivisorOnTropicalCurve(tc,coeffs)
+DivisorOnTropicalCurve{min, false}(An abstract tropical curve, [0, 1, 1, 1])
+
+julia> is_effective(dtc)
+true
+```
+"""
+is_effective(dtc::DivisorOnTropicalCurve{M, EMB}) where {M, EMB} = all(e -> e>=0, coefficients(dtc))
+
+
+@doc Markdown.doc"""
+   chip_firing_move(dtc::DivisorOnTropicalCurve{M, EMB}, position::Int)                
+
+Given a divisor `dtc` and vertex labelled `position`, compute the linearly equivalent divisor obtained by a chip firing move from the given vertex `position`. 
+```jldoctest
+julia> IM = IncidenceMatrix([[1,2],[1,3],[1,4],[2,3],[2,4],[3,4]]);
+
+julia> tc = TropicalCurve{min}(IM)
+An abstract tropical curve
+
+julia> coeffs = [0, 1, 1, 1];
+
+julia> dtc = DivisorOnTropicalCurve(tc,coeffs)
+DivisorOnTropicalCurve{min, false}(An abstract tropical curve, [0, 1, 1, 1])
+
+julia> chip_firing_move(dtc,1)
+DivisorOnTropicalCurve{min, false}(An abstract tropical curve, [-3, 2, 2, 2])
+```
+"""
+function chip_firing_move(dtc::DivisorOnTropicalCurve{M, EMB}, position::Int) where {M, EMB}
+    G = graph(base_curve(dtc))
+    newcoeffs = Vector{Int}(coefficients(dtc))
+    for i in 1:Polymake.nrows(G)
+        row = Polymake.row(G, i)
+        if position in row
+            newcoeffs[position] -= 1
+            for i in row
+                if i != position
+                    newcoeffs[i] += 1
+                end
+            end
+        end
+    end
+    return DivisorOnTropicalCurve(base_curve(dtc), newcoeffs)
+end
+
+### The function computes the outdegree of a vertex v  with respect to a given subset W  of vertices. 
+### This is the number of vertices not in W adjecent to v. 1,  
+function outdegree(tc::TropicalCurve{M,EMB}, W::Set{Int}, v::Int) where {M, EMB}
+    G = graph(tc)
+    m = Polymake.nrows(G) #number of edges of tc
+    @assert v in W "Vertex number $v not in $W"
+    deg = 0 #outdeg
+    for i in 1:m 
+        row = Polymake.row(G,i)
+        if v in row
+            for j in row
+                if !(j in W)
+                    deg = deg +1
+                end
+            end
+        end
     end
+    return deg
+end
+
+@doc Markdown.doc"""
+   v_reduced(dtc::DivisorOnTropicalCurve{M, EMB}, vertex::Int) 
```

Comment:
This is probably from the literature. if not, maybe `vertex_reduced` might be a more telling name. Feel free to ignore my comment :)

Reply:
It is from the literature. I will add a reference in the documentation.